src/bech32.cpp

```diff
@@ -27,6 +30,12 @@ const int8_t CHARSET_REV[128] = {
      1,  0,  3, 16, 11, 28, 12, 14,  6,  4,  2, -1, -1, -1, -1, -1
 };
 
+/* Determine the final constant to use for the specified encoding. */
+uint32_t EncodingConstant(Encoding encoding) {
```

Comment:
Is there a reason that this isn't simply a `std::map<Encoding, uint32_t>`?

This can be `constexpr`ed to evaluate this in compile time for the `VerifyChecksum()` case:

```suggestion
constexpr uint32_t EncodingConstant(Encoding encoding) {
```

Reply:
Hmm, a map feels like a lot more heavyweight/overkill to me.

Reply:
Making it `constexpr` will prevent backporting to pre-C++17 code. It's also unnecessary (constexpr doesn't control whether the compiler can evaluate at compile time as an optimization, only whether you can use it in compile-time expressions).